### DISCUSSION TITLE
SelfCollisionBarrier to avoid robotic arm hitting itself?

### DISCUSSION THREAD

**Discussion by MichaelRazum** [stephane-caron/pink#113] [Q&A] [answered]
Hi there,

Given a urdf file with specified collision. 

What would be the right approach, to avoid the robot hitting itself (or go through itself) ?  
Is using `SelfCollisionBarrier` the right way to do it?

I also tried to modify the code in **_examples/arm_ur5_.py** and adding  (line 52)
```python 
robot.collision_model.addAllCollisionPairs()
collision_barrier = SelfCollisionBarrier(
    n_collision_pairs=len(robot.collision_model.collisionPairs),
    gain=20.0,
    safe_displacement_gain=1.0,
    d_min=0.05,
)
barriers = [collision_barrier]
configuration = pink.Configuration(robot.model, robot.data, q_ref,
                                   collision_model=robot.collision_model,
                                   # Collision model is required for self_collision_barrier
                                   collision_data=robot.collision_data, )
```
I then added `barriers` to the solver. However, in this case, the process dies with exit code 139 (interrupted by signal 11: SIGSEGV). I suspect that the configuration is entirely wrong in the way I wrote it here.

Would be grateful for any hint!

**Comment by stephane-caron** [comment, marked as answer]
You can take a look at the self-collision examples in the examples/barriers directory. Those should be closer to what you are looking for here:

- [`kukas_self_collision.py`](https://github.com/stephane-caron/pink/blob/5a506b8d9ea2ac0d993c4857fab0e7fdf60a4736/examples/barriers/kukas_self_collision.py): whole-body self-collision avoidance
- [`yumi_self_collision.py`](https://github.com/stephane-caron/pink/blob/5a506b8d9ea2ac0d993c4857fab0e7fdf60a4736/examples/barriers/yumi_self_collision.py): self-collision avoidance with spheres

For a single-arm you can just skip the dual-arm duplication step at the beginning of these scripts.
